common/lib/xmodule/xmodule/conditional_module.py

```diff
@@ -262,4 +328,36 @@ def definition_to_xml(self, resource_fs):
         # Locations may have been changed to Locators.
         stringified_sources_list = map(lambda loc: loc.to_deprecated_string(), self.sources_list)
         self.xml_attributes['sources'] = ';'.join(stringified_sources_list)
+        self.xml_attributes[self.condional_attr] = self.conditional_value
+        self.xml_attributes['message'] = self.conditional_message
         return xml_object
+
+    def validate(self):
+        validation = super(ConditionalDescriptor, self).validate()
+        if not self.sources_list:
+            conditional_validation = StudioValidation(self.location)
+            conditional_validation.add(
+                StudioValidationMessage(
+                    StudioValidationMessage.NOT_CONFIGURED,
+                    _(u"This component has no source components configured yet."),
```

Comment:
I'm fine with this existing message. I think it's consistent with others in the platform, e.g. content libraries.